File: 1396-count-servers-that-communicate/1396-count-servers-that-communicate.java
class Solution {
    public int countServers(int[][] grid) {
        int res = 0;
        for(int i = 0;i < grid.length;i++){
            int sub = 0,ind = -1;
            for(int j = 0;j < grid[0].length;j++){
                if(grid[i][j] == 1){
                    sub++;
                    ind = j;
                }
            }
            if(sub > 1)
                res += sub;
            else if(ind != -1){
                for(int k = 0;k < grid.length;k++){
                    if(k != i && grid[k][ind] == 1){
                        res++;
                        break;
                    }
                }
            }
        }
        return res;
    }
}

File: 1396-count-servers-that-communicate/1396-count-servers-that-communicate.py
class Solution:
    def countServers(self, grid: List[List[int]]) -> int:
        res = 0
        for i in range(len(grid)):
            sub = 0
            ind = -1
            for j in range(len(grid[0])):
                if grid[i][j] == 1:
                    sub += 1
                    ind = j
            if sub > 1:
                res += sub
            elif ind != -1:
                for k in range(len(grid)):
                    if k != i and grid[k][ind] == 1:
                        res += 1
                        break
        return res